easybuild/framework/easyblock.py

```diff
@@ -2898,17 +2898,25 @@ def install_step_spec(initial):
         # part 2: iterated part, from 2nd iteration onwards
         # repeat core procedure again depending on specified iteration count
         # not all parts of all steps need to be rerun (see e.g., ready, prepare)
-        steps_part2 = [
-            ready_step_spec(False),
-            source_step_spec(False),
-            patch_step_spec,
-            prepare_step_spec,
-            configure_step_spec,
-            build_step_spec,
-            test_step_spec,
-            install_step_spec(False),
-            extensions_step_spec,
-        ] * (iteration_count - 1)
+        if multi_deps_extensions_only:
+            steps_part2 = [
+                ready_step_spec(False),
+                source_step_spec(False),
```

Comment:
@mboisson What problem did you encounter when kicking out `ready_step_spec` and `source_step_spec`?

Reply:
Well, for example, in the case of TensorRT, the extensions are actually bundled with the extracted binaries. So they need to be there. One of the steps removes them after the installation, so they need to be re-extracted. 

The ready_step is the one that handles having the different dependencies reloaded, so it's needed.

Reply:
I managed to get rid of the `source_step_spec` by modifying `ready_step_spec` conditionnaly to avoir recreating the build directory if `multi_deps_extensions_only` is used.